Translator.php

```diff
@@ -168,4 +175,27 @@ public function params()
     {
         return $this->_params;
     }
-} 
\ No newline at end of file
+
+    /**.
+     * @return array
+     */
+    public function filteredByFields():array
+    {
+        return $this->filteredByFields;
+    }
+
+    /**.
+     * @return bool
+     */
+    public function hasWhere():bool
```

Comment:
why this method?

Reply:
if (count($this->rules) > 0 && $translator->hasWhere()) {
            $filterQuery->where($translator->where())->addParams($translator->params());
        } else {
            $filterQuery->where('1=0');
        }

Reply:
if ```$this->rules``` is empty ```$translator->hasWhere()``` always will be ```false```

Reply:
in my case yes.  its double check
but without hasWhere   method where() returns empty string or '()'  when use groups
i have to check this options every time